pkg/utils/controller/controller.go

```diff
@@ -27,6 +28,20 @@ const (
 	labelSuccess      = "success"
 )
 
+var (
+	// ErrUnexpectedBehavior indicates the current situation is not expected.
+	// There should be something wrong with the system and cannot be recovered by itself.
+	ErrUnexpectedBehavior = errors.New("unexpected behavior which cannot be handled by the controller")
+)
+
+// NewUnexpectedBehaviorError returns ErrUnexpectedBehavior type error.
+func NewUnexpectedBehaviorError(err error) error {
+	if err != nil {
+		return fmt.Errorf("%w: %v", ErrUnexpectedBehavior, err.Error())
```

Comment:
https://pkg.go.dev/errors

Does it matter if %w is at the beginning?   will Is(err, ErrUnexpectedBehavior) return true for NewUnexpectedBehaviorError(err)